katz/src/test/kotlin/katz/OptionTest.kt

```diff
@@ -66,5 +66,32 @@ class OptionTest : UnitSpec() {
                     { value }
             ) shouldBe value
         }
+
+        "Option.monad.flatMap" should "be consistent with Option#flatMap" {
+            forAll { a: Int ->
+                val x = {b: Int -> Option(b * a)}
+                val option = Option(a)
+                option.flatMap(x) == Option.monad().flatMap(option, x)
+            }
+        }
+
+        "Option.monad.binding" should "for comprehend over option" {
+            val result = Option.monad().binding {
+                val x = ! Option(1)
+                val y = ! Option(1)
+                yields (x + y)
+            }
+            result shouldBe Option(2)
+        }
+
+        "Option.monad.binding" should "for comprehend over option" {
+            val result = Option.monad().binding {
+                val x = ! Option(1)
+                val y = ! Option(1)
+                yields (x + y)
+            }
+            result shouldBe Option(2)
+        }
```

Comment:
That's it then, out of a computer if you don't mind checking in a fix otherwise I'll change it tomorrow